core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,269 @@
+package io.metadew.iesi.openapi;
+
+import io.metadew.iesi.metadata.definition.component.Component;
+import io.metadew.iesi.metadata.definition.component.ComponentParameter;
+import io.metadew.iesi.metadata.definition.component.ComponentVersion;
+import io.metadew.iesi.metadata.definition.component.key.ComponentKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentParameterKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentVersionKey;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.media.Content;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import lombok.Data;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Data
+public class ComponentParser {
+    private static ComponentParser INSTANCE;
+
+    private Map<String, SecurityScheme> securitySchemeMap;
+    private Long versionNumber;
+    private String connectionName;
+
+
+    private ComponentParser() {}
+
+
+    public synchronized static ComponentParser getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ComponentParser();
+        }
+        return INSTANCE;
+    }
+
+    public List<Component> parse(OpenAPI openAPI) {
+        List<Component> components = new ArrayList<>();
+
+        securitySchemeMap = openAPI.getComponents().getSecuritySchemes();
+        versionNumber = Long.parseLong(openAPI.getInfo().getVersion());
+        connectionName = openAPI.getInfo().getTitle();
+
+        Paths paths = openAPI.getPaths();
+        for (String pathName : paths.keySet()) {
+            PathItem path = paths.get(pathName);
+            Map<PathItem.HttpMethod, Operation> operations = path.readOperationsMap();
+
+            for (Map.Entry<PathItem.HttpMethod, Operation> entry : operations.entrySet()) {
+                components.addAll(initComponents(entry.getValue(), entry.getKey(), pathName));
+            }
+
+        }
+
+       return components;
+    }
+
+    public List<Component> initComponents(Operation operation, PathItem.HttpMethod operationName, String pathName) {
+        List<Component> components = new ArrayList<>();
+        List<String> securities = new ArrayList<>(getSecurities(operation));
+        List<String> requestContents = new ArrayList<>(getRequestContents(operation.getRequestBody()));
+        List<String> responseContents = new ArrayList<>(getResponseContents(operation.getResponses()));
```

Comment:
Why convert the Set to List?

Reply:
Because on the generateNames method, I had to access element in the set. The issue is that getting an element from a set forces me to add more line in the method.

Reply:
Okay, but then why not return a list in getSecurities, getRequestContents and getResponseContents

Reply:
Indeed, I will update the methods to return List instead of Set